salesforce-pubsub/1.0/modules/ROOT/pages/salesforce-pubsub-connector-config-topics.adoc

```diff
@@ -0,0 +1,74 @@
+
+The event stream has a message durability of 72 hours for the high-volume platform events so, using the Replay ID, it is possible to receive a message from the stream within that period. 
+
+The MuleSoft subscriber must carefully track which Replay ID is processed to ensure it doesn't miss or duplicate process events in case of down time or restart (either intentional or unintentional). To enable advanced Replay ID tracking to ensure that the event is received and processed correctly, you are given control over the replay behavior with the Salesforce Pub/Sub Connector. 
+
+The following table shows the Replay ID options:
+
+[%header%autowidth.spread]
+|===
+| Replay Option | Description | Usage
+| ReplayId | The Subscriber receives all stored events after the event specified by its `replay_id` value, and new events. | Catch up on missed events after a certain event message, for example, after a connection failure. To subscribe with a specific Replay ID, save the Replay ID of the event message to retrieve stored events after, and then use it when you resubscribe.
+| `-1` | If no replay option is specified, this is the default. The Subscriber receives new events that are broadcast after the client subscribes. | It is best to subscribe with the `-1` option to receive new event messages. To get earlier event messages, don't use this option. 
+| `-2` | The Subscriber receives all events, including past events that are within the 72-hour retention window. | Catch up on missed events and retrieve all stored events, for example, after a connection failure. Using this option can slow performance if a large number of event messages are stored. 
+|===
+
+==== MuleSoft Object Store
+
+Mule runtime engine (Mule) uses object stores to persist data for later retrieval. Mule uses object stores in various filters, routers, and other message processors that need to store states between messages. 
+
+For CloudHub deployments, the free Object Store v2 limits usage to 10 transactions per second (TPS) per application. The premium add-on for Object Store v2 increases the limit to 100 TPS per app.
+
+=== Message Durability and Message Replay
+
+The Salesforce Event Bus stores messages for 72 hours. The receiver is responsible for keeping track of which messages are processed.
+
+It's best to design the pub-sub receiver flow so that it can reasonably recover after an intended or unintended restart. Specifically, this means that the receiver should be configured to pick up events from the last processed Replay Id only. The only way the receiver can get the Replay ID is by reading a key from the persistent object store. 
+
+=== Buffering the Standard Object Store for Higher Transaction Access
+
```

Comment:
This is not generally true, it applies in the context of the example shared here: https://docs.google.com/document/d/1NXLE93nzo7IRy4HSHsxh_InhhDJGv_m_xGo91VDyb3U/edit#bookmark=id.4w870phmjxv4 - that specific flow saves the replay id in the in-memory object store